iminuit/tests/test_iminuit.py

```diff
@@ -433,33 +433,68 @@ def f(x, y, z):
         m.migrad()
 
 
-class TestMatrix:
+class TestOutputInterface:
     def setup(self):
         self.m = Minuit(func3, print_level=0, pedantic=False)
         self.m.migrad()
+        self.m.hesse()
+        self.m.minos()
+
+    def test_args(self):
+        actual = self.m.args
+        expected = [2., 5.]
+        assert_allclose(actual, expected, atol=1e-8)
 
     def test_matrix(self):
         actual = self.m.np_matrix()
         expected = [[5., 0.], [0., 1.]]
-        assert_allclose(actual, expected)
+        assert_allclose(actual, expected, atol=1e-8)
+
+    def test_matrix_correlation(self):
+        actual = self.m.matrix(correlation=True)
+        expected = [[1., 0.], [0., 1.]]
+        assert_allclose(actual, expected, atol=1e-8)
 
     def test_np_matrix(self):
         import numpy as np
         actual = self.m.np_matrix()
         expected = [[5., 0.], [0., 1.]]
-        assert_allclose(actual, expected)
+        assert_allclose(actual, expected, atol=1e-8)
         assert isinstance(actual, np.ndarray)
 
-    def test_matrix_correlation(self):
-        actual = self.m.matrix(correlation=True)
-        expected = [[1., 0.], [0., 1.]]
-        assert_allclose(actual, expected)
+    def test_np_values(self):
+        import numpy as np
+        actual = self.m.np_values()
+        expected = [2., 5.]
+        assert_allclose(actual, expected, atol=1e-8)
+        assert isinstance(actual, np.ndarray)
+
+    def test_np_errors(self):
+        import numpy as np
+        actual = self.m.np_errors()
+        expected = [5.**0.5, 1.]
+        assert_allclose(actual, expected, atol=1e-8)
+        assert isinstance(actual, np.ndarray)
+
+    def test_np_merrors(self):
+        import numpy as np
+        actual = self.m.np_merrors()
+        expected = [[5.**0.5, 1.], [5.**0.5, 1.]]
+        assert_allclose(actual, expected, atol=1e-8)
+        assert isinstance(actual, np.ndarray)
```

Comment:
Can you please change this `isinstance` assert to an assert on `actual.shape`?
That tests a little bit more.
E.g. merrors is a 2-dim array, but values and errors should be 1-dim arrays, but from your current test would also pass if they were returned as 2-dim arrays with shape `(n_par, 1)`.
Looking at your implementation, they aren't, all is good.
But I think if it is easily possible to write tests that fully establish the behaviour without having to look at the implementation, that that is nice.

Reply:
Sure, good point.